src/WorkOS.net/Services/MFA/_interfaces/VerifyFactorResponse.cs

```diff
@@ -17,6 +17,18 @@ public class VerifyFactorResponse
         /// Validity of code of challenge.
         /// </summary>
         [JsonProperty("valid")]
-        public bool Valid { get; set; }
+        public bool IsValid { get; set; }
+
+        /// <summary>
+        /// Describe error code
+        /// </summary>
+        [JsonProperty("code")]
+        public string ErrorCode { get; set; }
```

Comment:
I don't think we should be mixing up our success responses and error responses.

It would be better to model these as two different types and then conditionally deserialize the response.

For example, if we get a successful status code we would deserialize into the success type, whereas a non-successful response code would be deserialized into the error type.

Reply:
The code key is only in the body response during 2 exception cases (already verified and expired). So to check for successful response would I just check for the keys in the response and then type cast them to the appropriate type? 

```
public async Task<Object> VerifyFactor(
            VerifyFactorOptions options,
            CancellationToken cancellationToken = default)
        {
            var request = new WorkOSRequest
            {
                Options = options,
                Method = HttpMethod.Post,
                Path = "/auth/factors/verify",
            };
            var response = await this.Client.MakeRawAPIRequest(request, cancellationToken).ConfigureAwait(false);
            var reader = new StreamReader(
                await response.Content.ReadAsStreamAsync().ConfigureAwait(false));
            var data = await reader.ReadToEndAsync().ConfigureAwait(false);
            var jsonObject = JObject.Parse(data);
            if (!jsonObject.ContainsKey("code") && jsonObject.ContainsKey("valid") && jsonObject.ContainsKey("challenge"))
            {
                return RequestUtilities.FromJson<VerifyFactorResponse>(data);
            }
            else if (jsonObject.ContainsKey("code") && !jsonObject.ContainsKey("valid") && !jsonObject.ContainsKey("challenge"))
            {
                return RequestUtilities.FromJson<VerifyFactorError>(data);
            }
            else
            {
                throw new Exception("Encountered unknown error, please reach out to support@workos.com.");
            }
        }
```

Reply:
I think we should be able to use the status code from the response to determine if we're getting back a valid response or an error.

So perhaps try something like this:

```cs
var response = await this.Client.MakeRawAPIRequest(request, cancellationToken).ConfigureAwait(false);
if (response.IsSuccessStatusCode)
{
    return response.Content.ReadAsJsonAsync<VerifyFactorResponse>();
}
else
{
    return response.Content.ReadAsJsonAsync<VerifyFactorErrorResponse>();
}
```

I'm not sure if we have `ReadAsJsonAsync` available, but if we don't then we could read the response body as a string and use this helper method: https://github.com/workos-inc/workos-dotnet/blob/03433892b46b775af4789db9731c7d1aa52ed908/src/WorkOS.net/Client/Utilities/RequestUtilities.cs#L82-L87